runtime/nvqir/cutensornet/mps_simulation_state.cpp

```diff
@@ -448,15 +430,27 @@ static Eigen::MatrixXcd reshapeStateVec(const Eigen::VectorXcd &stateVec) {
   return stacked;
 }
 
-std::unique_ptr<cudaq::SimulationState>
-MPSSimulationState::createFromSizeAndPtr(std::size_t size, void *ptr,
-                                         std::size_t dataType) {
-  Eigen::VectorXcd stateVec = Eigen::Map<Eigen::VectorXcd>(
-      reinterpret_cast<std::complex<double> *>(ptr), size);
+static Eigen::MatrixXcd reshapeStateVec(const Eigen::VectorXcd &stateVec) {
+  Eigen::MatrixXcd A = stateVec;
+  A.transposeInPlace();
+  return reshapeMatrix(A);
+}
+
+std::pair<std::unique_ptr<TensorNetState>, std::vector<MPSTensor>>
```

Comment:
Might be good to have a wrapper class that holds the unique_ptr.
```c++
struct MyTensorStateThingy {
  std::unique_ptr<TensorNetState> thing;
};
```
That way you can hide that there is a unique_ptr a bit from this interface. (Not sure it is worth it.)